File: Python/435无重复区间.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        n = len(intervals)
        if n == 0:
            return 0
        
        intervals.sort(key = lambda x:x[1])

        ans = 1
        right = intervals[0][1]

        for i in intervals[1:]:
            if i[0] >= right:
                right = i[1]
                ans += 1
        
        return n - ans

File: Java/435无重复区间.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        int n = intervals.length;
        if (n == 0)
            return 0;
        
        Arrays.sort(intervals, new Comparator<int[]>() {
            public int compare(int[] a, int[] b) {
                return a[1] - b[1];
            }
        });

        int right = intervals[0][1], ans = 1;
        for(int i=1; i<n; ++i) {
            if(intervals[i][0] >= right) {
                right = intervals[i][1];
                ++ans;
            }
        }
        return n - ans;
    }
}